platform/location_service.cpp

```diff
@@ -6,7 +6,11 @@
 #include <optional>
 #include <vector>
 
+#if defined(OMIM_OS_MAC)
 extern "C" location::LocationService * CreateAppleLocationService(location::LocationObserver &);
+#elif defined(OMIM_OS_LINUX)
+location::LocationService * CreateQtLocationService(location::LocationObserver &, std::string const & sourceName);
```

Comment:
Didn't we merge unique_ptr support?

Reply:
Yes we have, for `CreateDesktopLocationService` and this not that one, this is `CreateQtLocationService`.
Here the raw pointer is currently still needed because of `std::vector<LocationService *> m_services; ` has not been refactored to `std::vector<std::unique_ptr<LocationService>> m_services; ` yet.
I could do that too in a separate PR, if it is desired and there is capacity for it's review.

Reply:
Here is the PR for that: https://github.com/organicmaps/organicmaps/pull/5985